common.eai

```diff
@@ -12860,26 +12862,26 @@ function SingleMeleeAttackAM takes boolean needs_exp, boolean has_siege, boolean
       if needs_exp then
         if first_expansion_chosen and (current_expansion == null or CheckExpansionTaken(current_expansion)) then
           call ChooseExpansion()
-        endif
```

Comment:
This doesn't look right as the if statement will never allow an attack on expansions unless the current expansion is already claimed, which will cause them unable to expand until they randomly creep to the expansion instead.

Reply:
This can avoid the failure of expansion. Look at 3.1.1，or 2.6.1. This is exactly the same as that there. I firmly believe that I caused this error when I submitted it. This is just a restore

Reply:
but I will fix some err